File: pulsar-third/pulsar-chrome-devtools/src/main/java/com/github/kklisura/cdt/protocol/types/network/SignedExchangeSignature.java
package com.github.kklisura.cdt.protocol.types.network;

/*-
 * #%L
 * cdt-java-client
 * %%
 * Copyright (C) 2018 - 2019 Kenan Klisura
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import com.github.kklisura.cdt.protocol.support.annotations.Experimental;
import com.github.kklisura.cdt.protocol.support.annotations.Optional;
import java.util.List;

/**
 * Information about a signed exchange signature.
 * https://wicg.github.io/webpackage/draft-yasskin-httpbis-origin-signed-exchanges-impl.html#rfc.section.3.1
 */
@Experimental
public class SignedExchangeSignature {

  private String label;

  private String signature;

  private String integrity;

  @Optional private String certUrl;

  @Optional private String certSha256;

  private String validityUrl;

  private Integer date;

  private Integer expires;

  @Optional private List<String> certificates;

  /** Signed exchange signature label. */
  public String getLabel() {
    return label;
  }

  /** Signed exchange signature label. */
  public void setLabel(String label) {
    this.label = label;
  }

  /** The hex string of signed exchange signature. */
  public String getSignature() {
    return signature;
  }

  /** The hex string of signed exchange signature. */
  public void setSignature(String signature) {
    this.signature = signature;
  }

  /** Signed exchange signature integrity. */
  public String getIntegrity() {
    return integrity;
  }

  /** Signed exchange signature integrity. */
  public void setIntegrity(String integrity) {
    this.integrity = integrity;
  }

  /** Signed exchange signature cert Url. */
  public String getCertUrl() {
    return certUrl;
  }

  /** Signed exchange signature cert Url. */
  public void setCertUrl(String certUrl) {
    this.certUrl = certUrl;
  }

  /** The hex string of signed exchange signature cert sha256. */
  public String getCertSha256() {
    return certSha256;
  }

  /** The hex string of signed exchange signature cert sha256. */
  public void setCertSha256(String certSha256) {
    this.certSha256 = certSha256;
  }

  /** Signed exchange signature validity Url. */
  public String getValidityUrl() {
    return validityUrl;
  }

  /** Signed exchange signature validity Url. */
  public void setValidityUrl(String validityUrl) {
    this.validityUrl = validityUrl;
  }

  /** Signed exchange signature date. */
  public Integer getDate() {
    return date;
  }

  /** Signed exchange signature date. */
  public void setDate(Integer date) {
    this.date = date;
  }

  /** Signed exchange signature expires. */
  public Integer getExpires() {
    return expires;
  }

  /** Signed exchange signature expires. */
  public void setExpires(Integer expires) {
    this.expires = expires;
  }

  /** The encoded certificates. */
  public List<String> getCertificates() {
    return certificates;
  }

  /** The encoded certificates. */
  public void setCertificates(List<String> certificates) {
    this.certificates = certificates;
  }
}

File: pulsar-core/pulsar-tools/pulsar-browser/src/main/kotlin/ai/platon/cdt/kt/protocol/types/network/SignedExchangeSignature.kt
@file:Suppress("unused")
package ai.platon.cdt.kt.protocol.types.network

import ai.platon.cdt.kt.protocol.support.annotations.Experimental
import ai.platon.cdt.kt.protocol.support.annotations.Optional
import com.fasterxml.jackson.`annotation`.JsonProperty
import kotlin.Int
import kotlin.String
import kotlin.collections.List

/**
 * Information about a signed exchange signature.
 * https://wicg.github.io/webpackage/draft-yasskin-httpbis-origin-signed-exchanges-impl.html#rfc.section.3.1
 */
@Experimental
data class SignedExchangeSignature(
  @param:JsonProperty("label")
  val label: String,
  @param:JsonProperty("signature")
  val signature: String,
  @param:JsonProperty("integrity")
  val integrity: String,
  @param:JsonProperty("certUrl")
  @param:Optional
  val certUrl: String? = null,
  @param:JsonProperty("certSha256")
  @param:Optional
  val certSha256: String? = null,
  @param:JsonProperty("validityUrl")
  val validityUrl: String,
  @param:JsonProperty("date")
  val date: Int,
  @param:JsonProperty("expires")
  val expires: Int,
  @param:JsonProperty("certificates")
  @param:Optional
  val certificates: List<String>? = null,
)
